### PR TITLE
Feature/video javascript integration

### PR DESCRIPTION
**Pull request by danbudris**
This branch was created in order to merge the WebRTC javascript previously present in videochat.html into the main comm_tools.js file, and to modify the css and html to allow the in-browser presentation of the video without the use of iframe (and the corresponding https header hackiness).  Tested working in production.

### CODE REVIEW

File: group1/comm/static/comm_tool.js [hunk 1]
```diff
@@ -825,3 +823,230 @@ function downloadFileCallback(data) {
...
+        for (peer_id in peers) {
+            peers[peer_id].close();
+        }
+
+        peers = {};
+        peer_media_elements = {};
+    });
+    function join_chat_channel(channel, userdata) {
+        video_socket.emit('join', {"channel": channel, "userdata": userdata});
+        //alert("join channel");
```
Comment (gggoddardbu) [inline group1/comm/static/comm_tool.js]: Did you want to keep this commented line?

File: group1/comm/static/comm_tool.js [hunk 2]
```diff
@@ -825,3 +823,230 @@ function downloadFileCallback(data) {
...
+    }
+    function part_chat_channel(channel) {
+        video_socket.emit('part', channel);
+    }
+
+   // video_socket.emit("join", {"channel": default_channel, "userdata":userdata});
```
Comment (gggoddardbu) [inline group1/comm/static/comm_tool.js]: Did you want to keep this line?

File: group1/comm/static/comm_tool.js [hunk 3]
```diff
@@ -825,3 +823,230 @@ function downloadFileCallback(data) {
...
+
+    video_socket.on("addPeer", function (config) {
+       // alert("recieving add PEER");
```
Comment (gggoddardbu) [inline group1/comm/static/comm_tool.js]: Can this line be removed?

File: group1/comm/static/comm_tool.js [hunk 4]
```diff
@@ -825,3 +823,230 @@ function downloadFileCallback(data) {
...
+        console.log('Signaling server said to add peer:', config);
+        //alert(config);
```
Comment (gggoddardbu) [inline group1/comm/static/comm_tool.js]: Can this line be removed?

File: group1/comm/static/comm_tool.js [hunk 5]
```diff
@@ -825,3 +823,230 @@ function downloadFileCallback(data) {
...
+            }
+        }
+        peer_connection.onaddstream = function(event) {
+            console.log("onAddStream", event);
+            var remote_media = USE_VIDEO ? $("<video>") : $("<audio>");
+            remote_media.attr("autoplay", "autoplay");
+            if (MUTE_AUDIO_BY_DEFAULT) {
+                remote_media.attr("muted", "true");
+            }
+            //remote_media.attr("controls", "");
```
Comment (gggoddardbu) [inline group1/comm/static/comm_tool.js]: Are lines 910-913 still needed?

File: group1/comm/static/comm_tool.js [hunk 6]
```diff
@@ -825,3 +823,230 @@ function downloadFileCallback(data) {
...
+    });
+    video_socket.on('sessionDescription', function(config) {
+        console.log('Remote description received: ', config);
+        var peer_id = config.peer_id;
+        var peer = peers[peer_id];
+        var remote_description = config.session_description;
+        console.log(config.session_description);
+
+        var desc = new RTCSessionDescription(remote_description);
+        var stuff = peer.setRemoteDescription(desc,
```
Comment (gggoddardbu) [inline group1/comm/static/comm_tool.js]: Can var stuff be renamed to something more indicative of what it is?  Maybe "remote_description_promise"?

File: group1/comm/static/comm_tool.js [hunk 7]
```diff
@@ -825,3 +823,230 @@ function downloadFileCallback(data) {
...
+    };
+
+    navigator.getUserMedia({"audio":USE_AUDIO, "video":USE_VIDEO},
+        function(stream) { /* user accepted access to a/v */
+            console.log("Access granted to audio/video");
+            local_media_stream = stream;
+            var local_media = USE_VIDEO ? $("<video>") : $("<audio>");
+            local_media.attr("autoplay", "autoplay");
+            local_media.attr("muted", "true"); /* always mute ourselves by default */
+            //local_media.attr("controls", "");
```
Comment (gggoddardbu) [inline group1/comm/static/comm_tool.js]: Can lines 1037-1039 be removed?

**Comment by askarbayev** [comment]
Sure, you can remove all commented lines. I just wanted to check what is
going on in every step. You can also change the variable name if it is not
enough to be clear for reading or understanding code.

File: group1/comm/static/comm_tool.js (hunk 1 above)
Comment (danbudris) [reply group1/comm/static/comm_tool.js]: nope. resolved.

File: group1/comm/static/comm_tool.js (hunk 2 above)
Comment (danbudris) [reply group1/comm/static/comm_tool.js]: nope, resolved.

File: group1/comm/static/comm_tool.js (hunk 3 above)
Comment (danbudris) [reply group1/comm/static/comm_tool.js]: removed

File: group1/comm/static/comm_tool.js (hunk 5 above)
Comment (danbudris) [reply group1/comm/static/comm_tool.js]: removed

File: group1/comm/static/comm_tool.js (hunk 7 above)
Comment (danbudris) [reply group1/comm/static/comm_tool.js]: fixed

File: group1/comm/static/comm_tool.js (hunk 6 above)
Comment (danbudris) [reply group1/comm/static/comm_tool.js]: adjusted

**Comment by danbudris** [comment]
OK, updated as per GG

**Pull request merged**